### PR TITLE
feat: [FC-0047] FCM

### PR DESCRIPTION
**Pull request by volodymyr-chekyrta**
This pull request introduces Firebase Cloud Messaging (FCM) to the project, enabling the functionality for receiving push notifications from Firebase.

Testing:
• Verified push notifications are received when the app is in the foreground, background, and terminated states.
• Confirmed the app correctly handles notification taps and displays relevant content.
• Ensured token generation and handling work as expected, including handling token refreshes.

Please use the Firebase console for testing; here is a set of all the required fields for navigation:

Notifications:
1. User has been enrolled
notification_type: `enroll`
course_id: Your course ID (example: course-v1:future+f1+2024)
2. User has been unenrolled
notification_type: `unenroll`
3. User has been added as a beta tester
notification_type: `add_beta_tester`
course_id: Your course ID (example: course-v1:future+f1+2024)
4. User has been removed from beta testers
notification_type: `remove_beta_tester`
course_id: Your course ID (example: course-v1:future+f1+2024)
5. Someone left a response in the user discussion/question
notification_type: `forum_response`
topic_id: Your Topic ID (example: course)
course_id: Your course ID (example: course-v1:future+f1+2024)
thread_id: Your thread ID (example: 66584e96f2195c001ad219c7)
comment_id: Your comment ID (example: 66585c9df2195c001ad219ce)
6. Someone left a comment in the user discussion/question
notification_type: `forum_comment`
topic_id: Your Topic ID (example: course)
course_id: Your course ID (example: course-v1:future+f1+2024)
thread_id: Your thread ID (example: 66583d13f2195c001a56a182)
parent_id: Your response ID (example: 66586579f2195c001ad219d8)
comment_id: Your comment ID (example: 6659ac92f2195c001bf37f77)

### CODE REVIEW

**Comment by volodymyr-chekyrta** [comment]
@dixidroid @omerhabib26, Kindly review the changes made in this pull request.
Thank you 🙌

**Review by dixidroid — DISMISSED** [review]

**Comment by dixidroid** [comment]
LGTM

**Review by dixidroid — DISMISSED** [review]

**Comment by HamzaIsrar12** [comment]
Reviewing ...

**Review by HamzaIsrar12 — CHANGES_REQUESTED** [review]
Minor nits only. 👍🏻

File: app/src/main/java/org/openedx/app/data/api/NotificationsApi.kt [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+interface NotificationsApi {
+
+    @POST("/api/mobile/v4/notifications/create-token/")
+    @FormUrlEncoded
+    suspend fun syncFirebaseToken(
+        @Field("registration_id") token: String,
+        @Field("active") active: Boolean = true
+    )
+
```
[the remark this answers is not in the document]
Comment (volodymyr-chekyrta) [reply app/src/main/java/org/openedx/app/data/api/NotificationsApi.kt]: Sure ✅

File: app/src/main/java/org/openedx/app/AppViewModel.kt [hunk 2]
```diff
@@ -42,20 +52,40 @@ class AppViewModel(
 
     override fun onCreate(owner: LifecycleOwner) {
         super.onCreate(owner)
-        setUserId()
+
```
[the remark this answers is not in the document]
Comment (volodymyr-chekyrta) [reply app/src/main/java/org/openedx/app/AppViewModel.kt:55]: Good suggestion 👍

File: app/src/main/java/org/openedx/app/system/push/OpenEdXFirebaseMessagingService.kt [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+            notificationManager.createNotificationChannel(channel)
+        }
+
+        notificationManager.notify(code, notificationBuilder.build())
+    }
+
+    private fun createId(): Int {
+        return SystemClock.uptimeMillis().toInt()
+    }
+
```
[the remark this answers is not in the document]
Comment (volodymyr-chekyrta) [reply app/src/main/java/org/openedx/app/system/push/OpenEdXFirebaseMessagingService.kt]: ✅

File: app/src/main/java/org/openedx/app/deeplink/DeepLink.kt [hunk 4]
```diff
@@ -3,20 +3,57 @@ package org.openedx.app.deeplink
 class DeepLink(params: Map<String, String>) {
 
     val screenName = params[Keys.SCREEN_NAME.value]
+    val notificationType = params[Keys.NOTIFICATION_TYPE.value]
```
[the remark this answers is not in the document]
Comment (volodymyr-chekyrta) [reply app/src/main/java/org/openedx/app/deeplink/DeepLink.kt]: ✅

**Comment by volodymyr-chekyrta** [comment]
@HamzaIsrar12, thanks for your feedback, PR is ready for the next pass 🚀

**Review by HamzaIsrar12 — APPROVED** [review]
LGTM 🏎️

**Comment by openedx-webhooks** [comment]
@volodymyr-chekyrta 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/openedx-app-android&entry.867055334=https://github.com/openedx/openedx-app-android/pull/344&entry.1484655318=https://github.com/volodymyr-chekyrta&entry.752974735=2024-06-18+08:46&entry.1917517419=2024-06-25+10:32&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**